source/utils/paragraphHelper.py

```diff
@@ -0,0 +1,251 @@
+		line = tempTi.text.strip()
+		if len(line) > 0:
+			paragraph += line + " "
+		if isLastLineOfParagraph(tempTi.text):
+			break
+		if not tempTi.move(textInfos.UNIT_LINE, 1):
+			break
+		numLines += 1
+
+	if len(paragraph.strip()) > 0:
+		chunks = splitParagraphIntoChunks(paragraph)
+		for chunk in chunks:
+			speech.speakMessage(chunk)
+	else:
+		# Translators: This is spoken when a paragraph is considered blank.
+		speech.speakMessage(_("blank"))
+
+
+def moveToParagraph(nextParagraph: bool, speakNew: bool) -> tuple((bool, bool)):
+	"""
+	Moves to the previous or next normal paragraph, delimited by a single line break.
+	@param nextParagraph: bool indicating desired direction of movement,
+	True for next paragraph, False for previous paragraph
+	@param speakNew: bool indicating if new paragraph should be spoken after navigating
+	@returns: A boolean 2-tuple of:
+	-passKey: if True, should send the gesture on
+	- moved: if True, position has changed
+	"""
+	ti = getTextInfoAtCaret()
+	if (ti is None) or (not isAcceptableTextInfo(ti)):
+		return (True, False)
+	ti.expand(textInfos.UNIT_LINE)
+	ti.collapse()  # move to start of line
+	moveOffset = 1 if nextParagraph else -1
+	moved = False
+	numLines = 0
+	tempTi = ti.copy()
+	tempTi.expand(textInfos.UNIT_LINE)
+	# if starting line is the last line of a paragraph, move back two paragraph markers
+	moveBackTwice = isLastLineOfParagraph(tempTi.text)
+	while numLines < MAX_LINES:
+		tempTi = ti.copy()
+		tempTi.expand(textInfos.UNIT_LINE)
+		if isLastLineOfParagraph(tempTi.text):
+			if not nextParagraph:
+				if not moveBackTwice:
+					while numLines < MAX_LINES:
+						if not ti.move(textInfos.UNIT_LINE, -1):
+							moved = True  # pin to beginning
+							break
+						tempTi = ti.copy()
+						tempTi.expand(textInfos.UNIT_LINE)
+						if isLastLineOfParagraph(tempTi.text):
+							ti.move(textInfos.UNIT_LINE, 1)
+							moved = True
+							break
+						numLines += 1
+
+					break  # break out of outer while loop
+				else:
+					moveBackTwice = False
+			else:  # moving to next paragraph
+				if ti.move(textInfos.UNIT_LINE, 1):
+					moved = True
+				break
+		if not ti.move(textInfos.UNIT_LINE, moveOffset):
+			break
+		numLines += 1
+
+	if moved:
+		ti.updateCaret()
+		if isinstance(ti, UIATextInfo):
+			ti._rangeObj.ScrollIntoView(False)
+		speakParagraph(ti)
+	else:
+		tones.beep(1000, 30)
+	return (False, moved)
+
+
+def speakBlockParagraph(ti: textInfos.TextInfo) -> None:
+	paragraph = ""
+	numLines = 0
+	tempTi = ti.copy()
+	while numLines < MAX_LINES:
+		tempTi.expand(textInfos.UNIT_LINE)
+		line = tempTi.text.strip()
+		if not len(line):
+			break
+		paragraph += line + "\r\n"
+		if not tempTi.move(textInfos.UNIT_LINE, 1):
+			break
+		numLines += 1
+
+	chunks = splitParagraphIntoChunks(paragraph)
+	for chunk in chunks:
+		speech.speakMessage(chunk)
+
+
+def moveToBlockParagraph(
+		nextParagraph: bool, speakNew: bool, ti: textInfos.TextInfo = None) -> tuple((bool, bool)):
```

Comment:
nextParagraph: bool, speakNew: bool, ti: textInfos.TextInfo = None) -> Tuple[bool, bool]:
This will not run, get following error at runtime:
TypeError: 'type' object is not subscriptable

Reply:
are you importing `Tuple` from `typing`?